src/SynapseWriter.php

```diff
@@ -0,0 +1,386 @@
+
+        return !empty($res);
+    }
+
+    private function execQuery(string $query): void
+    {
+        $this->logger->info(sprintf("Executing query '%s'", $this->hideCredentialsInQuery($query)));
+        try {
+            $this->db->exec($query);
+        } catch (\Throwable $e) {
+            throw new UserException('Query execution error: ' . $e->getMessage(), 0, $e);
+        }
+    }
+
+    private function fetchAll(string $query): array
+    {
+        /** @var \PDOStatement $stmt */
+        $stmt = $this->db->query($query);
+        /** @var array $result */
+        $result = $stmt->fetchAll(PDO::FETCH_ASSOC);
+        return $result;
+    }
+
+
+    public function testConnection(): void
+    {
+        $this->execQuery('SELECT 1');
+    }
+
+    public function generateTmpName(string $tableName): string
+    {
+        // "#" is a mark for the temp table
+        return rtrim(
+            mb_substr(
+                sprintf(
+                    '#%s_%s',
+                    self::TEMP_NAME,
+                    str_replace('.', '_', $tableName)
+                ),
+                0,
+                255
+            ),
+            '-'
+        );
+    }
+
+    public function generateStageName(string $tableName): string
+    {
+        return rtrim(
+            mb_substr(
+                sprintf(
+                    '%s_%s',
+                    self::STAGE_NAME,
+                    str_replace('.', '_', $tableName)
+                ),
+                0,
+                255
+            ),
+            '-'
+        );
+    }
+
+    public function getCurrentUser(): string
+    {
+        return $this->fetchAll('SELECT CURRENT_USER;')[0]['CURRENT_USER'];
+    }
+
+
+    private function getColumnsSqlDefinition(array $table): string
+    {
+        $columns = array_filter((array) $table['items'], function ($item) {
+            return (strtolower($item['type']) !== 'ignore');
+        });
+
+        $sql = '';
+
+        foreach ($columns as $col) {
+            $type = strtoupper($col['type']);
+            if (!empty($col['size']) && in_array(strtolower($col['type']), self::$typesWithSize)) {
+                $type .= sprintf('(%s)', $col['size']);
+            }
+            $null = $col['nullable'] ? 'NULL' : 'NOT NULL';
+            $default = empty($col['default']) ? '' : "DEFAULT '{$col['default']}'";
+            if ($type === 'TEXT') {
+                $default = '';
+            }
+            $sql .= sprintf(
+                '%s %s %s %s,',
+                self::quoteIdentifier($col['dbName']),
+                $type,
+                $null,
+                $default
+            );
+        }
+
+        return trim($sql, ' ,');
+    }
+
+    private function hideCredentialsInQuery(string $query): ?string
+    {
+        return preg_replace(
+            '/(SECRET=)\'[^\']+\'/i',
+            '$1\'...\'',
+            $query
+        );
+    }
+
+    public function validateTable(array $tableConfig): void
```

Comment:
Tu sa moze validovat ci ma existujuca tabulka dane stlpce. 
No nerobi sa to ani v `db-writer-snowflake`.